server/server.go

```diff
@@ -58,17 +61,45 @@ func Start(port string) {
 	log.Fatal(http.ListenAndServe(port, r))
 }
 
-//	determins the hostname to return based on the following hierarchy
-//	- HostName var as configured via the config file
-//	- The request host
+//	determines the hostname:port to return based on the following hierarchy
+//	- HostName / Port vars as configured via the config file
+//	- The request host / port if config HostName or Port is missing
 func hostName(r *http.Request) string {
-	//	configured
+	var requestHostname string
+	var requestPort string
+	substrs := strings.Split(r.Host, ":")
+	switch len(substrs) {
+	case 1:
+		requestHostname = substrs[0]
+	case 2:
+		requestHostname = substrs[0]
+		requestPort = substrs[1]
+	default:
+		log.Printf("Multiple colons (':') in host string: %v", r.Host)
+	}
+
+	var retHost string
 	if HostName != "" {
-		return HostName
+		retHost = HostName
+	} else {
+		retHost = requestHostname
+	}
+
+	// Strip colon from Port if it is present.
+	portNumber := Port
+	if len(portNumber) > 0 && string([]rune(portNumber[0:1])) == ":" {
+		portNumber = portNumber[1:]
+	}
+
+	if portNumber == "none" {
+		// Don't add a port to the host.
+	} else if portNumber != "" {
+		retHost += ":" + portNumber
+	} else if requestPort != "" {
+		retHost += ":" + requestPort
 	}
```

Comment:
I would rewrite this `if else` statement to:
```
if portNumber != "" && portNumber != "none" {
      return retHost + ":" + portNumber
}
if requestPort != "" {
     return retHost + ":" + portNumber
}
return retHost
```

Reply:
Don't see any benefit to this change, but will make.